core_types/src/agent.rs

```diff
@@ -73,16 +76,17 @@ impl AgentId {
     /// `s` will be used for the `nick` and included in the key string as well
     /// this agent is not cryptographically generated...
     /// it will not be able to sign / encrypt anything
-    pub fn generate_fake(s: &str) -> Self {
-        let mut buf = s.to_string();
-        while buf.len() < 84 {
+    pub fn generate_fake(nick: &str) -> Self {
```

Comment:
This method obviously won't be used in production, so I think it would be good to add a
`#[cfg(test)]` attribute or don't event define it inside the `AgentId` impl

Reply:
But because we don't have ways yet to handle real agent keys this is only way to create a syntactically correct agent id object. It is used in the container and I just used it in the the ContextBuilder: https://github.com/holochain/holochain-rust/pull/761

I would suggest to leave at as it is for now and do what you suggest once we have reduced its usage to just testing..

Reply:
agreed @lucksus